panoptes/routes/api.py

```diff
@@ -70,12 +70,14 @@ def get_job_of_workflow(workflow_id, job_id):
 
 @routes.route('/api/delete/<workflow_id>', methods=['GET'])
 def set_db_delete(workflow_id):
-    if(get_db_workflows_by_id(workflow_id)!=None):
+    if(get_db_workflows_by_id(workflow_id)==None):
+        return jsonify({'error': 404, 'msg': 'Unable to delete Workflow ' + workflow_id + 
+                        '. Please check if workflow ' + workflow_id + ' exists.'})
+    elif(get_db_workflows_by_status(workflow_id)=='Running'):
```

Comment:
Maybe the best is to call `get_db_workflows_by_status(workflow_id)` once before the if statements. Now you call the same thing twice.

Reply:
We have 2 deferment errors. I thing is better to check if wf is on database and after check if his status is Running.